src/compiler/transformers/esDecorators.ts

```diff
@@ -0,0 +1,2087 @@
+                    const varDecls = factory.createVariableDeclarationList([varDecl], NodeFlags.Let);
+                    const statement = factory.createVariableStatement(modifiers, varDecls);
+                    setOriginalNode(statement, node);
+                    setCommentRange(statement, getCommentRange(node));
+                    return statement;
+                }
+            }
+            else {
+                const modifiers = visitNodes(node.modifiers, node => tryCast(node, isModifier));
+                const heritageClauses = visitNodes(node.heritageClauses, visitor, isHeritageClause);
+                enterClass(/*classInfo*/ undefined);
+                const members = visitNodes(node.members, classElementVisitor, isClassElement);
+                exitClass();
+                return factory.updateClassDeclaration(node, modifiers, node.name, /*typeParameters*/ undefined, heritageClauses, members);
+            }
+        }
+
+        function visitClassExpression(node: ClassExpression, referencedName: Expression | undefined) {
+            if (isDecoratedClassLike(node)) {
+                const className = node.name ? factory.createStringLiteralFromNode(node.name) : referencedName ?? factory.createStringLiteral("");
+                const iife = transformClassLike(node, className);
+                setOriginalNode(iife, node);
+                return iife;
+            }
+            else {
+                const modifiers = visitNodes(node.modifiers, node => tryCast(node, isModifier), isModifier);
+                const heritageClauses = visitNodes(node.heritageClauses, visitor, isHeritageClause);
+                enterClass(/*classInfo*/ undefined);
+                const members = visitNodes(node.members, classElementVisitor, isClassElement);
+                exitClass();
+                return factory.updateClassExpression(node, modifiers, node.name, /*typeParameters*/ undefined, heritageClauses, members);
+            }
+        }
+
+        function prepareConstructor(parent: ClassLikeDeclaration, classInfo: ClassInfo) {
+            if (classInfo.instanceExtraInitializersName && !classInfo.hasNonAmbientInstanceFields) {
+                // If there are instance extra initializers we need to add them to the body along with any
+                // field initializers
+                const statements: Statement[] = [];
+
+                statements.push(
+                    factory.createExpressionStatement(
+                        emitHelpers().createRunInitializersHelper(
+                            factory.createThis(),
+                            classInfo.instanceExtraInitializersName
+                        )
+                    )
+                );
+
+                for (const member of parent.members) {
+                    if (isPropertyDeclaration(member) && !isStatic(member)) {
+                        // TODO(rbuckton)

```

Comment:
when is this planned to be done?

Reply:
It looks like most of the function is left over from stubbing out a place to handle ES decorators on ambient fields, but I've opted to hold off on that since it requires significant additional work to support it for `--target esnext` in a way that aligns with the runtime semantics of other decorators. In choosing to postpone that work, the `ts` transform currently elides ambient fields (and an error is reported by the checker). As a result, when entering this function you either don't have instance fields at all (in which case we never get to that line), or you do have instance fields but none of them are ambient (so the entire function body is elided).

I'll clean this up in the next commit.

Reply:
I think this is also partially related to `accessor` field transformations, prior to my moving them into the `classFields` transform.